### PR TITLE
Add bandit workflow

### LINKED ISSUE #781: Add bandit workflow
### Problem Description

_As a developer, it would be useful to get a static code analysis of our library every so often so we aren't accidently introducing known vulnerabilities._

### Expected behavior

- Add a Github Actions workflow that runs when a release is made. This action should
    - Run Bandit
    - Store the output as a file at the base level of the repo
- Make sure the file doesn't get included when creating the package for RDT

### Additional context
- See [this PR](https://github.com/sdv-dev/SDV/pull/1888) for inspiration
[Bandit ](https://bandit.readthedocs.io/en/latest/)is an open sourced tool that can be used to scan python code for vulnerabilities.

### LINKED ISSUE #783: Fix minimum version workflow when pointing to github branch
### Error Description

The minimum version workflow crashes when a library points to a Github branch.
The fix should allow us to run the minimum version workflow with libraries pointing to Github branches.

### PR DESCRIPTION
**Pull request by R-Palazzo**
CU-86azzkgrf
Resolve #781
Resolve #783

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86azzkgrf RDT - Add bandit workflow #781](https://app.clickup.com/t/86azzkgrf)

File: .github/workflows/static_code_analysis.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install bandit==1.7.7
+    - name: Save code analysis
+      run: bandit -r . -x ./tests -f txt -o static_code_analysis.txt --exit-zero
+    - name: Create pull request
+      id: cpr
+      uses: peter-evans/create-pull-request@v4
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (fealho) [inline .github/workflows/static_code_analysis.yml:27]: I think we are switching these to GH_ACCESS_TOKEN, right?

Comment (R-Palazzo) [reply .github/workflows/static_code_analysis.yml]: Good catch, done in [c3be2fa](https://github.com/sdv-dev/RDT/pull/784/commits/c3be2fa467978b2013d348a67c24f9d8cc0850ec)

**Review by fealho — APPROVED** [review]
👍

**Review by rwedge — APPROVED** [review]

**Pull request merged**